### ISSUE TITLE
Can’t verify address with Ledger Nano X

### ISSUE THREAD

**Issue by Coolman916** [Emurgo/yoroi-frontend#2249]
I’m using Yoroi wallet on iPhone and it’s the latest version 4.2.0. I can’t verify the receiving address on my Ledger Nano X. The error message says: no addressing data, should not happen. I have double checked my Bluetooth connection and I updated the Cardano app on ledger to 2.4.1. I also double checked that the ledger firmware is up to date 1.3.0. Yoroi wallet has worked with my Nano X perfectly up until this point.

**Comment by roadrunr** [comment]
Same problem I am facing with Ledger Nano S - I am not using Bluetooth but a cable for connection.
It's really quite alarming to experience this connectivity failure and would be nice if someone was assigned to address these issues we are experiencing!

**Comment by Clos88** [comment]
I have the same issue with my nano X . I have the latest software update for both. It will not  connect to verify address “no addressing data ,should not happen” . I have had no problems in the past but now I can’t verify my ledger. I have even removed my wallet, went to add wallet- connect to ledger nano. It recognizes my nano x and I am able to recover my wallet, however, when I go to verify address I still same issue where it says “ no addressing data….” 
Please help

**Comment by zuzunker** [comment]
Hi, @Clos88 and @roadrunr! I am sorry to hear you are experiencing such a problem! We are investigating this issue at the moment. We have started seeing some users having issues with their Ledger Nano models specifically, even tho we did not make any changes to the main hardware-wallet flow in the meantime. Some users have managed to get the issue resolved by getting the Cardano app updated in their Ledger Live, but I see that you have already done this. It seems like there might have been some unexpected changes to Nano models specifically that somehow cause problems with out implementation. 

We will try to get some details on that as soon as possible and see if can resolve it quickly!

**Comment by SpeedRacer-7** [comment]
I’m having the same problem.

**Comment by TheoMer** [comment]
I’m experiencing the very same issue.

‘An connection error occurred while trying to connect with your hardware wallet. Error: No addressing data, should never happen.’

Ledger Nano X firmware - 1.3.0   
iPhone - 6s plus (ios 14.7.1).     

Prior to the ledger firmware 1.3.0 update I was able to unstake (deregister), now I get an ‘undefined is not an object (evaluating ‘p.stake credentials’). 

I mention the unstake issue as I believe the issue is related to being unable to verify addresses. I also get an connection error message when attempting to unstake as well.

Bluetooth is on on iOS. The ada app has been opened on ledger. But an connection error is generated.

What’s the issue here?

**Comment by Scapa1919** [comment]
Same issue here on iPhone XS, latest iOS. Cannot stake, send or authenticate a receive adress. Removing the Ledger Wallet and re-installing it works, though.

**Comment by TheoMer** [comment]
@Scapa1919 

> ... Removing the Ledger Wallet and re-installing it works, though.

Can you explain the step by step process you used to re-install the wallet? Will selecting 'Connect to Ledger Nano' and going through the process create a new wallet with the outstanding ADA balance as held (referenced to) on my Ledger Nano X?

**Comment by Scapa1919** [comment]
If I delete the Wallet on Yoroi and re-intsall it, the wallet appears with all funds in it, so no problem here. But when I try to send ADA, it apparently not receive the private key from the Nano X and hence cannot verify the transaction.

**Comment by iamnido** [comment]
@Scapa1919 ah. you mean, those step (removing Yoroi wallet and adding it again) works but not necessarily fix the issue. correct?

**Comment by Scapa1919** [comment]
yes, right. Removing and re-installing works, but it does not solve the issue with the transaction verification through the Nano X

**Comment by TheoMer** [comment]
@Scapa1919 Ah. So I guess it's back to awaiting an Emurgo resolution on this issue. Thanks for the heads up.

**Comment by Scapa1919** [comment]
no problem. I’ll just wait. Only annoying thing is that I cannot stake at the moment.

**Comment by MajorPain14** [comment]
I am also having this problem. I'm also not able to restore the hardware wallet on my desktop, it seems. I am getting a network error, but there is nothing wrong with my network.

**Comment by AdVanceYourself** [comment]
Just wanted to post that I submitted a case on the Telegram for a similar error being unable to perform transactions. The error was "invalid asset group in the token bundle in an output - incorrect ordering of tokens"
For anyone else that needs an alternative, adalite.io has been working without issue in the meantime.

**Comment by ghost** [comment]
I have the same issue here, using a fully updated Ledger Nano X on a Bluetooth connection.

**Comment by TheoMer** [comment]
@vsubhuman Can we expect to see a resolution to this issue on/by the 12th September when the Alonzo hard fork takes place?

**Comment by juraskrlec** [comment]
Is there any status on this? The workaround is to use Chrome extension, which is working fine but still having issues on iOS. Using Ledger 1.3.0 and Cardano 2.4.1

**Comment by TheoMer** [comment]
Any idea when this issue is going to be resolved?

**Comment by dylansweb2021** [comment]
Verify address works perfectly with Ledger when I use the Yoroi chrome extension but installed in the Brave Browser.
In the Yoroi Android app, and in the Firefox extension, I get the "Connection" error when I try to verify my receive address on the Ledger device.

Using the chrome Yoroi extension installed in Brave was the first time verifying an address on my Ledger device worked.
However, when I export the public key from the device it mentions "Byron" and I believed it would be "Shelly"...not sure if this will matter or not - but at least I did finally verify my receive address using the Ledger hardware wallet.

**Comment by TheoMer** [comment]
I'm still experiencing this issue with:

1. Nano X firmware 2.0.0
2. Yoroi version 4.4.0
3. iOS 15.0.2

What is the issue here?

**Comment by SpeedRacer-7** [comment]
I’m using a iOS device.  Will this work?


Chris

**Comment by TheoMer** [comment]
> I’m using a iOS device.  Will this work?
> 
> 
> Chris
> 
> > On Sep 27, 2021, at 12:40 AM, dylansweb2021 ***@***.***> wrote:
> > 
> > ﻿
> > Verify address works perfectly with Ledger when I use the Yoroi chrome extension but installed in the Brave Browser.
> > In the Yoroi Android app, and in the Firefox extension, I get the "Connection" error when I try to verify my receive address on the Ledger device.
> > 
> > Using the chrome Yoroi extension installed in Brave was the first time verifying an address on my Ledger device worked.
> > However, when I export the public key from the device it mentions "Byron" and I believed it would be "Shelly"...not sure if this will matter or not - but at least I did finally verify my receive address using the Ledger hardware wallet.
> > 
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub, or unsubscribe.
> 

It's not working for me.

**Comment by advanture-space** [comment]
I had the same issue. Couldn't verify address. All that while ledger live mobile didn't have any issues whatsoever.

The problem with this issue is that on mobile we're kinda blind. There's no debug data that can be easily viewed whatsoever. This is issue number 1. Could you please add a way to dump the full state of the app (probably without sensitive data)?

After re-installation and re-pairing address verification worked.

So this is an issue of statefulness. The state of the app matters. Before and after re-installation. We need to dump these and compare these.

What I would suggest:

1. add the ability to dump the state of the app, some kind of log and all the non-sensitive variables
2. while having this issue, dump all the state
3. after re-installation and not having this issue, dump all the state again
4. compare the two states to see what's the difference which should then give a pretty good idea what's wrong

**Comment by jarstorm** [comment]
Hello, and thank you for your contribution.
We're currently reviewing long-standing issues to keep the tracker manageable. As this issue hasn't seen activity in some time, we're closing it.
If the problem persists or you'd like to follow up, feel free to comment and we'll reopen it.
